src/RFC-0181_BulletproofsPlus.md

```diff
@@ -0,0 +1,160 @@
+CONTRACT, STRICT LIABILITY OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+## Language
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT
+RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in
+[BCP 14](https://tools.ietf.org/html/bcp14) (covering RFC2119 and RFC8174) when, and only when, they appear in all
+capitals, as shown here.
+
+## Disclaimer
+
+This document and its content are intended for information purposes only and may be subject to change or update without
+notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community of the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+This Request for Comment (RFC) describes Tari-specific implementation details for Bulletproofs+ range proving and verifying.
+
+## Introduction
+
+The Tari implementation of the [Bulletproofs+](https://eprint.iacr.org/2020/735) range proving system makes several changes and optimizations that we describe here.
+In particular, it supports the following useful features:
+- Commitments can be _extended_; that is, they can commit to a value using multiple masks.
+- Range proofs can be _aggregated_; that is, a single range proof can assert valid range for multiple commitments in an efficient way.
+- A set of arbitrary range proofs can be verified in a _batch_; that is, the verifier can check the validity of all proofs in the set at once in an efficient way.
+- The prover can assert a nonzero minimum value bound to a commitment.
+- The prover can delegate to certain verifiers the ability to recover the masks used for the extended commitment in a non-aggregated proof.
+
+The Bulletproofs+ preprint does not address extended commitments, as it only defines its range proving system using Pedersen commitments.
+However, a later preprint for [Zarcanum](https://eprint.iacr.org/2021/1478) updates the algorithms and security proofs to accommodate one additional mask, and the reasoning extends generally.
+Aggregation of range assertions using Pedersen commitments is described in the Bulletproofs+ preprint, and the Zarcanum preprint describes the corresponding changes for extended commitments.
+Batch verification is described only informally in the Bulletproofs+ preprint, and in an incomplete fashion.
+Minimum value assertion is not addressed in the preprint.
+An approach to mask and value recovery was [used by Grin](https://github.com/mimblewimble/grin-wallet/issues/105) for a different range proving system, and can be modified to support Bulletproofs+ range proofs with extended commitments.
```

Comment:
We can also reference the deprecated RFC-0180 here as the previous Tari implementation for mask and value recovery

Reply:
What's the correct way to deprecate RFC-0180?